test/e2e/tests/request-queuing/dapp1-switch-dapp2-send.spec.js

```diff
@@ -165,4 +165,158 @@ describe('Request Queuing Dapp 1, Switch Tx -> Dapp 2 Send Tx', function () {
       },
     );
   });
+
+  it('should queue send tx after switch network confirmation and transaction should target the correct network after switch is cancelled.', async function () {
```

Comment:
I think it would be helpful if there were a comment above this describing that the original bug was noticed on https://app.odos.xyz/, describing the manual procedure for reproducing, and pointing to the PRs that fix it.

Reply:
this test is covering a different scenario from the `eth_requestAccount` requests not being passed through that you discovered with ODOS

Instead see: https://github.com/MetaMask/metamask-extension/pull/25310